heron/common/src/cpp/network/baseserver.cpp

```diff
@@ -106,6 +106,17 @@ sp_int32 BaseServer::Start_Base() {
     return -1;
   }
 
+  // fetch the port after bind/listen port 0
+  if (AF_INET == options_.get_sin_family() && 0 == options_.get_port()) {
+    if (getsockname(listen_fd_, serv_addr, &sockaddr_len) != 0)  {
```

Comment:
This code appears only once.
This PR is the prerequisite of the other PRs. If this PR is merged, this code will not appear in the other PRs.